lib/wisper/activejob/broadcaster.rb

```diff
@@ -13,9 +13,39 @@ def broadcast(subscriber, _publisher, event, args)
       def perform?(subscriber, event, args)
         # If a predicate method of the format "event_name?" is defined on the subscriber, only run
         # the subscriber if it evaluates to true
-        return subscriber.send("#{event}?", *args) if subscriber.respond_to?("#{event}?")
+        return subscriber.send("perform_#{event}?", *args) if subscriber.respond_to?("perform_#{event}?")
 
         true
+      rescue StandardError
+        raise DispatchError.new(subscriber, event, args)
+      end
+
+      class DispatchError < StandardError
+        def initialize(subscriber, event, args)
+          super("Unable to perform #{name(subscriber)}##{event}(#{args_info(args)})")
+        end
+
+        private
+
+        # Copied from LoggerBroadcaster
+        def name(object)
+          id_method  = %w(id uuid key object_id).find do |method_name|
+            object.respond_to?(method_name) && object.method(method_name).arity <= 0
+          end
+          id         = object.send(id_method)
+          class_name = object.class == Class ? object.name : object.class.name
+          "#{class_name}##{id}"
+        end
+
+        def args_info(args)
+          return if args.empty?
+
+          args.map do |arg|
+            arg_string = name(arg)
+            arg_string += ": #{arg.inspect}" if [Numeric, Array, Hash, String].any? {|klass| arg.is_a?(klass) }
+            arg_string
+          end.join(', ')
+        end
```

Comment:
I'm missing the context for this change.  What is the purpose of rescuing and rethrowing the error here?  Isn't this swallowing the original error and losing the context?

Reply:
This is just to make sure we don't fail without a way to rerun the job we would have previously fired off. Today, if we create a job for a listener method and it fails, sidekiq retry logic protects us. With the previous change to this gem, we'd never create the job and we'd lose the information needed to recreate it.

The `cause` of the new `DispatchError` should be the error we're rescuing. I'll have to double check that that's good enough.

Reply:
```
irb(main):083:1* def foo
irb(main):084:2*   begin
irb(main):085:2*     raise StandardError.new('test')
irb(main):086:2*   rescue StandardError => e
irb(main):087:2*     raise NoMethodError.new('othertest')
irb(main):088:1*   end
irb(main):089:0> end
=> :foo
irb(main):090:0> foo
(irb):87:in `rescue in foo': othertest (NoMethodError)
	from (irb):83:in `foo'
	from (irb):90:in `<main>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `<main>'
(irb):85:in `foo': test (StandardError)
	from (irb):90:in `<main>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `<main>'
irb(main):091:1* begin
irb(main):092:1*   foo
irb(main):093:1* rescue StandardError => e
irb(main):094:1*   puts e.inspect
irb(main):095:1*   puts e.cause.inspect
irb(main):096:0> end
#<NoMethodError: othertest>
#<StandardError: test>
=> nil
```

Reply:
I'm pretty sure the whole exception chain gets logged, so we shouldn't be losing any info.